### ISSUE TITLE
Windows Checks causing exceptions in sensor

### ISSUE THREAD

**Issue by thozook** [StackStorm-Exchange/stackstorm-icinga2#13]
While setting up Stackstorm with this pack, I noticed that my test rule was not being enforced. Upon troubleshooting the sensor by running
```
/opt/stackstorm/st2/bin/st2sensorcontainer --config-file=/etc/st2/st2.conf --sensor-ref=icinga2.Icinga2StateChangeSensor
```
as root from the command line, I found that Icinga Windows checks were causing an exception to be thrown in the sensor and the trigger is not dispatched.

Exception Example 1:
```
2020-09-17 10:20:52,526 INFO [-] Connecting to event stream API.
2020-09-17 10:20:52,718 INFO [-] Processing event: {"check_result":{"active":true,"check_source":"###HOSTNAME###","command":["C:\Program Files\ICINGA2\/sbin/check_load.exe","-c","95","-w","90"],"execution_end":1600363364.508,"execution_start":1600363362.94,"exit_status":2.0,"output":"LOAD CRITICAL 97.28% ","performance_data":["load=97.28%;90;95;0;100"],"schedule_end":1600363364.512,"schedule_start":1600363364.512,"state":2.0,"ttl":0.0,"type":"CheckResult","vars_after":{"attempt":1.0,"reachable":true,"state":2.0,"state_type":0.0},"vars_before":{"attempt":1.0,"reachable":true,"state":0.0,"state_type":1.0}},"host":"###HOSTNAME###","service":"Windows Load","state":2.0,"state_type":0.0,"timestamp":1600363363.52031,"type":"StateChange"}
2020-09-17 10:20:52,719 ERROR [-] Icinga2StateChangeSensor Exception Invalid \escape: line 1 column 92 (char 91)
Traceback (most recent call last):
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 66, in run
    self.process_event(event)
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 76, in process_event
    event = json.loads(event)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Invalid \escape: line 1 column 92 (char 91)
2020-09-17 10:20:57,721 INFO [-] Connecting to event stream API.
```

Exception Example 2:
```
2020-09-17 10:20:52,526 INFO [-] Connecting to event stream API.
2020-09-17 10:20:52,718 INFO [-] Processing event: {"check_result":{"active":true,"check_source":"###HOSTNAME###","command":["C:\Program Files\ICINGA2\/sbin/check_load.exe","-c","95","-w","90"],"execution_end":1600363364.508,"execution_start":1600363362.94,"exit_status":2.0,"output":"LOAD CRITICAL 97.28% ","performance_data":["load=97.28%;90;95;0;100"],"schedule_end":1600363364.512,"schedule_start":1600363364.512,"state":2.0,"ttl":0.0,"type":"CheckResult","vars_after":{"attempt":1.0,"reachable":true,"state":2.0,"state_type":0.0},"vars_before":{"attempt":1.0,"reachable":true,"state":0.0,"state_type":1.0}},"host":"###HOSTNAME###","service":"Windows Load","state":2.0,"state_type":0.0,"timestamp":1600363363.52031,"type":"StateChange"}
2020-09-17 10:20:52,719 ERROR [-] Icinga2StateChangeSensor Exception Invalid \escape: line 1 column 92 (char 91)
Traceback (most recent call last):
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 66, in run
    self.process_event(event)
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 76, in process_event
    event = json.loads(event)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Invalid \escape: line 1 column 92 (char 91)
2020-09-17 10:20:57,721 INFO [-] Connecting to event stream API.
```

**Comment by thozook** [comment]
Tried a workaround where I replaced the traditional windows backwards slashes with forwards slashes along with using the newer Icinga Powershell Framework(https://github.com/Icinga/icinga-powershell-framework) plus their Powershell plugins(https://github.com/Icinga/icinga-powershell-plugins) since that is what Icinga will be using for Windows checks going forward, but ended up running into another issue with the data from the event:
```
2020-09-18 12:37:25,986 INFO [-] Processing event: {"check_result":{"active":true,"check_source":"###HOSTNAME###","command":["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe","-command","Use-Icinga; exit Invoke-IcingaCheckService","-Service","###SERVICENAME###","-Status","Running","-Verbosity","0","-NoPerfData","False"],"execution_end":1600457959.922,"execution_start":1600457950.89,"exit_status":2.0,"output":"[CRITICAL] Check package "Services" - [CRITICAL] Service "###PRETTY SERVICE NAME### (###SERVICENAME###)"
\_ [CRITICAL] Service "###PRETTY SERVICE NAME### (###SERVICENAME###)": Value "Stopped" is not matching threshold "Running"","performance_data":[],"schedule_end":1600457959.922,"schedule_start":1600457959.922,"state":2.0,"ttl":0.0,"type":"CheckResult","vars_after":{"attempt":1.0,"reachable":true,"state":2.0,"state_type":0.0},"vars_before":{"attempt":1.0,"reachable":true,"state":0.0,"state_type":1.0}},"host":"###HOSTNAME###","service":"###ICINGA CHECK NAME###","state":2.0,"state_type":0.0,"timestamp":1600457959.642696,"type":"StateChange"}
2020-09-18 12:37:25,988 ERROR [-] Icinga2StateChangeSensor Exception Expecting ',' delimiter: line 1 column 411 (char 410)
Traceback (most recent call last):
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 66, in run
    self.process_event(event)
  File "/opt/stackstorm/packs/icinga2/sensors/icinga2_state_change_events.py", line 76, in process_event
    event = json.loads(event)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Expecting ',' delimiter: line 1 column 411 (char 410)
2020-09-18 12:37:30,990 INFO [-] Connecting to event stream API.
```

**Comment by nmaludy** [comment]
Sorry, i'm not super familiar with this pack and don't have Icinga to test against. @thozook would you be willing to try and help fix this with a Pull Request?

**Comment by thozook** [comment]
I created #14 to address some of the issues that come up with the sensor and the JSON object(such as the presences of backwards slashes and control characters). I'm not sure how to deal with extra quotation marks being in the string unfortunately, but this should provide some relief for people.

I managed to deal with the issues of quotation marks in the string that are not escaped by switching my checks over to the Icinga Powershell Framework and then editing the New-IcingaCheck.psm1 and New-IcingaCheckPackage.psm1 files to remove the quotation marks from the check output.
